1.20.4 Error on some components: `Invalid color name: reset`

Command that triggers this error:
```
/tellraw @p {"text":"Lorem ","extra":[{"text":"Ipsum","italic":true}]}
```


```java
[11:55:14 ERROR]: [Triton] Unhandled exception occurred in onAsyncPacket() for Triton
java.lang.IllegalStateException: Unable to invoke method public static net.minecraft.network.chat.IChatMutableComponent net.minecraft.network.chat.IChatBaseComponent$ChatSerializer.b(java.lang.String)
        at com.comphenix.protocol.reflect.accessors.DefaultMethodAccessor.invoke(DefaultMethodAccessor.java:25) ~[ProtocolLib.jar:?]
        at com.comphenix.protocol.wrappers.WrappedChatComponent.deserialize(WrappedChatComponent.java:73) ~[ProtocolLib.jar:?]
        at com.comphenix.protocol.wrappers.WrappedChatComponent.fromJson(WrappedChatComponent.java:109) ~[ProtocolLib.jar:?]
        at com.rexcantor64.triton.packetinterceptor.ProtocolLibListener.handleSystemChat(ProtocolLibListener.java:326) ~[Triton.jar:?]
        at com.rexcantor64.triton.packetinterceptor.ProtocolLibListener.onPacketSending(ProtocolLibListener.java:749) ~[Triton.jar:?]
        at com.comphenix.protocol.async.AsyncListenerHandler.processPacket(AsyncListenerHandler.java:620) ~[ProtocolLib.jar:?]
        at com.comphenix.protocol.async.AsyncListenerHandler.listenerLoop(AsyncListenerHandler.java:576) ~[ProtocolLib.jar:?]
        at com.comphenix.protocol.async.AsyncListenerHandler.access$100(AsyncListenerHandler.java:49) ~[ProtocolLib.jar:?]
        at com.comphenix.protocol.async.AsyncListenerHandler$1.run(AsyncListenerHandler.java:221) ~[ProtocolLib.jar:?]
        at com.comphenix.protocol.async.AsyncListenerHandler.lambda$start$1(AsyncListenerHandler.java:290) ~[ProtocolLib.jar:?]
        at org.bukkit.craftbukkit.v1_20_R3.scheduler.CraftTask.run(CraftTask.java:101) ~[paper-1.20.4.jar:git-Paper-390]
        at org.bukkit.craftbukkit.v1_20_R3.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:57) ~[paper-1.20.4.jar:git-Paper-390]
        at com.destroystokyo.paper.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:22) ~[paper-1.20.4.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: com.google.gson.JsonParseException: Invalid color name: reset
        at net.minecraft.Util.getOrThrow(Util.java:882) ~[?:?]
        at net.minecraft.network.chat.Component$Serializer.deserialize(Component.java:291) ~[?:?]
        at net.minecraft.network.chat.Component$Serializer.fromJsonLenient(Component.java:325) ~[?:?]
        at com.comphenix.protocol.reflect.accessors.DefaultMethodAccessor.invoke(DefaultMethodAccessor.java:23) ~[ProtocolLib.jar:?]
        ... 15 more
```

--- a/core/src/main/java/com/rexcantor64/triton/language/parser/AdvancedComponent.java
+++ b/core/src/main/java/com/rexcantor64/triton/language/parser/AdvancedComponent.java
@@ -170,6 +170,9 @@ public class AdvancedComponent {
                     continue;
                 }
                 ChatColor previousColor = ComponentUtils.getColorFromBaseComponent(component);
+                if (previousColor == ChatColor.RESET) {
+                    previousColor = null;
+                }
                 if (builder.length() != 0) {
                     component.setText(builder.toString());
                     builder = new StringBuilder();

